gt/main.c

```diff
@@ -705,55 +706,62 @@ alloc_and_fill_notify_pkt(unsigned int socket, struct ggu_policy *policy,
 
 	/* Fill up the policy decision. */
 	memset(notify_ggu, 0, sizeof(*notify_ggu));
-	notify_ggu->v1 = GGU_PD_VER1;
+	notify_ggu->version = GGU_PD_VER;
 	if (policy->flow.proto == ETHER_TYPE_IPv4
 			&& policy->state == GK_DECLINED) {
-		notify_ggu->n1 = 1;
-		data = (uint8_t *)rte_pktmbuf_append(notify_pkt,
-			sizeof(policy->flow.f.v4) +
-			sizeof(policy->params.u.declined));
-		rte_memcpy(data, &policy->flow.f.v4,
+		decision_len += sizeof(policy->flow.f.v4) +
+			sizeof(policy->params.u.declined);
+		ggu_decision = (struct ggu_decision *)
+			rte_pktmbuf_append(notify_pkt, decision_len);
+		ggu_decision->type = GGU_DEC_IPV4_DECLINED;
+		rte_memcpy(ggu_decision->ip_flow, &policy->flow.f.v4,
 			sizeof(policy->flow.f.v4));
-		rte_memcpy(data + sizeof(policy->flow.f.v4),
+		rte_memcpy(ggu_decision->ip_flow + sizeof(policy->flow.f.v4),
```

Comment:
We must replace the `rte_memcpy()` here and in the cases that follow for calls to `rte_cpu_to_be_32()`. I know that you have not introduced this bug, but we still need to fix it.